src/modules/message.js

```diff
@@ -47,32 +50,43 @@ class Message {
     @return {object} - Prefix for the log message. Or an empty string of no prefix data logging is enabled
   **/
   getPrefix(settings = {}, delimiter = ':') {
-    if (this.prefix === null) {
-      const message = this.payload;
-      const includeTimestamp = process.env.LOG_TIMESTAMP === 'true' || !!settings.LOG_TIMESTAMP;
-      const includeEnvironment = process.env.LOG_ENVIRONMENT === 'true' || !!settings.LOG_ENVIRONMENT;
-      const includeLogLevel = process.env.LOG_LEVEL === 'true' || !!settings.LOG_LEVEL;
-      const environment = process.env.NODE_ENV === 'undefined' ? 'local' : process.env.NODE_ENV;
-      this.prefix = {
-        timestamp: includeTimestamp ? `${new Date().toISOString()}${delimiter}` : '',
-        environment: includeEnvironment ? `${environment}${delimiter}` : '',
-        logLevel: includeLogLevel ? `${message.level.toUpperCase()}${delimiter}` : '',
-        reqId: ''
-      };
-      if (message.meta.reqId) {
-        const includeReqId = process.env.LOG_REQID === 'true' || !!settings.LOG_REQID;
-        this.prefix.reqId = includeReqId ? `${message.meta.reqId}` : '';
-      }
-      let isEmpty = true;
-      for (const key in this.prefix) { // eslint-disable-line
-        if (this.prefix[key] !== '') {
-          isEmpty = false;
-          break;
-        }
+    const message = this.payload;
+    const includeTimestamp = process.env.LOG_TIMESTAMP ?
+      process.env.LOG_TIMESTAMP === 'true' : !!settings.LOG_TIMESTAMP;
+    const includeEnvironment = process.env.LOG_ENVIRONMENT ?
+      process.env.LOG_ENVIRONMENT === 'true' : !!settings.LOG_ENVIRONMENT;
+    const includeLogLevel = process.env.LOG_LEVEL ?
+      process.env.LOG_LEVEL === 'true' : !!settings.LOG_LEVEL;
+    const environment = process.env.NODE_ENV === undefined ?
+      'local' : process.env.NODE_ENV;
+    const includeCallerPrefix = process.env.LOG_CALLER_PREFIX ?
+      process.env.LOG_CALLER_PREFIX === 'true' : !!settings.LOG_CALLER_PREFIX;
```

Comment:
### Suggestions
Since the same logic is applied on all those variables, maybe extract the logic as a function